toolkits/slack/arcade_slack/utils.py

```diff
@@ -444,3 +450,44 @@ def convert_relative_datetime_to_unix_timestamp(
     days, hours, minutes = map(int, relative_datetime.split(":"))
     seconds = days * 86400 + hours * 3600 + minutes * 60
     return int(current_unix_timestamp - seconds)
+
+
+def instantiate_api_error(slack_api_error: SlackApiError) -> ThirdPartyApiError:
+    error_message = slack_api_error.response.get("error", str(slack_api_error))
+    retry_after_ms = get_retry_after_ms(slack_api_error.response.headers)
+    http_response = HttpResponse(
+        status_code=slack_api_error.response.status_code,
+        headers=slack_api_error.response.headers,
+        body=str(slack_api_error.response.data),
+    )
+
+    if http_response.status_code >= 500:
+        error = ThirdPartyApiServerError(
+            http_response=http_response,
+            error_message=error_message,
+            retry_after_ms=retry_after_ms,
+        )
+    elif http_response.status_code == 429:
+        error = ThirdPartyApiRateLimitError(
+            http_response=http_response,
+            error_message=error_message,
+            retry_after_ms=retry_after_ms,
+        )
+    else:
+        error = ThirdPartyApiError(
+            http_response=http_response,
+            error_message=error_message,
+        )
```

Comment:
cc @EricGustin - I'll bet we can come up with a utility that makes this part part simple too, including a common form for `get_retry_after_ms`